### DISCUSSION TITLE
古老的courses作业的疑问

### DISCUSSION THREAD

**Discussion by Fangyu-Lu** [physics-data/faq#144] [Q&A] [answered]
运行grade.py的时候报错是依赖错误，好像是说每一个课程的最后一个依赖课程多了后缀，而且只有最后一个报错。

我单独make thesis查看了一下生成的文件里面课程都是没有后缀的，不知道哪里出了问题？

![0c79cc3def6d405e1231b09a6871f83](https://user-images.githubusercontent.com/77487856/132177355-9e8881f5-b4e5-4e30-8258-36b2ba3f2ecf.png)
![efa9f5e9b25890ac67e29578281a3bc](https://user-images.githubusercontent.com/77487856/132177397-43f2723d-89d7-454e-9193-f5f7fc8238ed.png)

**Comment by heroxbd** [comment]
@Harry-Chen

**Comment by Harry-Chen** [comment, marked as answer]
你的脚本的确有问题，`SUFFIX=.test make thesis` 就能看到问题了。你之所以没看出来问题是因为自己跑的时候没有传入 `SUFFIX`，当然是空的。

**Reply by Harry-Chen** [reply]
通过运行 `make -n` 很容易就能发现问题
